### DISCUSSION TITLE
Unconditionally concatenate sheets within .xlsx workbook?

### DISCUSSION THREAD

**Discussion by hoclun-rigsep** [saulpw/visidata#1097] [Q&A] [answered]
How do this? In other words if the .xlsx contains x sheets with y total rows, I would like to ignore column headers and create a sheet with y rows—no join logic at all.

**Comment by saulpw** [comment, marked as answer]
If I understand correctly, you would load the sheets with `--header 0` and then use the `append` jointype to combine all the sheets by rows.

**Reply by hoclun-rigsep** [reply]
Okay, that is what I thought and it works fine on a minimal example (I'm running develop branch). I must have something specific to my workbook going on, because `append` is doing weird things there. Thanks for responding.

**Reply by saulpw** [reply]
Sure thing!  If you want to set up a videochat, we can look at the workbook that's not cooperating and see what's going on.  It might be something trivial, or it might be an actual bug!  I have some ideas but it's hard to say without seeing the actual data and output.

**Comment by geekscrapy** [comment]
Can someone share the answer to this please :)

I am mostly interested in how you join inner sheets that appear in IndexSheets of the multiple xlsx files. Not figured a way to do this yet

**Reply by hoclun-rigsep** [reply]
From the man page:

```
Sheets Sheet (Shift+S)
     open Sheets Stack, which contains only the active sheets on the current stack
     (global commands)
        gS               open Sheets Sheet, which contains all sheets from current session, active and inactive
        Alt number       jump to sheet number
     (sheet-specific commands)
         Enter           jump to sheet referenced in current row
        gEnter           push selected sheets to top of sheet stack
         a               add row to reference a new blank sheet
        gC  gI           open Columns Sheet/Describe Sheet with all visible columns from selected sheets
        g^R              reload all selected sheets
        z^C  gz^C        abort async threads for current/selected sheets(s)
        g^S              save selected or all sheets
         & jointype      merge selected sheets with visible columns from all, keeping rows according to jointype:
                         .  inner  keep only rows which match keys on all sheets
                         .  outer  keep all rows from first selected sheet
                         .  full   keep all rows from all sheets (union)
                         .  diff   keep only rows NOT in all sheets
                         .  append keep all rows from all sheets (concatenation)
                         .  extend copy first selected sheet, keeping all rows and sheet type, and extend with columns from other sheets
                         .  merge  mostly keep all rows from first selected sheet, except prioritise cells with non-null/non-error values
```

You want the `&` operator with the "append" jointype.

**Reply by geekscrapy** [reply]
Sorry, I misread the original question...🤦‍♂️ I thought you were trying to join sheets from multiple xlsx files, rather than from one xlsx file 👍
